local-cli/runIOS/RunIOSInternals.js

```diff
@@ -0,0 +1,272 @@
+/**
+* Copyright (c) 2015-present, Facebook, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree. An additional grant
+* of patent rights can be found in the PATENTS file in the same directory.
+*/
+'use strict';
+
+const child_process = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const findXcodeProject = require('./findXcodeProject');
+const parseIOSDevicesList = require('./parseIOSDevicesList');
+const findMatchingSimulator = require('./findMatchingSimulator');
+
+function RunIOSInternals(os_device,
+                         os_simulator,
+                         os_scheme_suffix,
+                         os_type,
+                         os_run_name,
+                         os_default_simulator) {
+  this.os_device = os_device;
+  this.os_simulator = os_simulator;
+  this.os_scheme_suffix = os_scheme_suffix;
+  this.os_type = os_type;
+  this.os_run_name = os_run_name;
+  this.os_default_simulator = os_default_simulator;
+}
+
+RunIOSInternals.prototype.getBuildPath = function(configuration = 'Debug', appName, isDevice) {
+  return `build/Build/Products/${configuration}-${isDevice ? this.os_device : this.os_simulator}/${appName}.app`;
+};
+
+RunIOSInternals.prototype.runIOS = function(argv, config, args) {
+  process.chdir(args.projectPath);
+  const xcodeProject = findXcodeProject(fs.readdirSync('.'));
+  if (!xcodeProject) {
+    throw new Error('Could not find Xcode project files in ios folder');
+  }
+
+  const inferredSchemeName = path.basename(xcodeProject.name, path.extname(xcodeProject.name)) + this.os_scheme_suffix;
+  const scheme = args.scheme || inferredSchemeName;
+  console.log(`Found Xcode ${xcodeProject.isWorkspace ? 'workspace' : 'project'} ${xcodeProject.name}`);
+  const devices = parseIOSDevicesList(
+    child_process.execFileSync('xcrun', ['instruments', '-s'], {encoding: 'utf8'})
+  );
+  if (args.device) {
+    const selectedDevice = this.matchingDevice(devices, args.device);
+    if (selectedDevice){
+      return this.runOnDevice(selectedDevice, scheme, xcodeProject, args.configuration, args.packager);
+    } else {
+      if (devices){
+        console.log('Could not find device with the name: "' + args.device + '".');
+        console.log('Choose one of the following:');
+        this.printFoundDevices(devices);
+      } else {
+        console.log('No ' + this.os_type + ' devices connected.');
+      }
+    }
+  } else if (args.udid) {
+    return this.runOnDeviceByUdid(args, scheme, xcodeProject, devices);
+  } else {
+    return this.runOnSimulator(xcodeProject, args, scheme);
+  }
+};
+
+RunIOSInternals.prototype.runOnDeviceByUdid = function(args, scheme, xcodeProject, devices) {
```

Comment:
Again, I believe all these should go out of prototype to normal functions.

Something like:
```
function runOnDeviceByUdid() {}
function runOnDevice() {}
function matchingDeviceByUdid() {}

module.exports = function makeRunCommand() {
   return () => {
     // this function runs command
   };
}
```